### PR TITLE
Return Book Control

### PR DESCRIPTION
**Pull request by elliotmitchum**
Refactors `ReturnBookControl` class.

Criteria:

- Evaluate against the class diagram
- Evaluate against the code guidelines
- Project builds and runs

### CODE REVIEW

**Comment by amrmosit** [comment]
@haynescrossley this branch is ready for review. 
The code is still working and operational. (checked)

File: src/library/returnBook/ReturnBookControl.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package library.returnBook;
+import library.entities.Item;
+import library.entities.Library;
+import library.entities.Loan;
+
+public class ReturnBookControl {
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookControl.java]: #PR41 - D1: 
Please rename class name/file name to ReturnItemControl as in the BRLS class diagram.

File: src/library/returnBook/ReturnBookControl.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    private ReturnBookUI ui;
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookControl.java]: #PR41 - D2: 
Please rename/refactor variable to ReturnItemUI as in the BRLS class diagram.

File: src/library/returnBook/ReturnBookControl.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    private enum ControlState { INITIALISED, READY, INSPECTING };
+
+    private ControlState state;
+    
+    private Library library;
+
+    private Loan currentLoan;
+    
+
+    public ReturnBookControl() {
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookControl.java]: #PR41 - D3: 
Constructor isn't listed in class diagram.. though please rename to ReturnItemControl.

File: src/library/returnBook/ReturnBookControl.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            throw new RuntimeException("ReturnBookControl: cannot call setUI except in INITIALISED state");
+        }
+
+        this.ui = ui;
+        ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void bookScanned(long bookId) {
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookControl.java]: #PR41 - D4: 
Please rename method to itemScanned as in the BRLS class diagram.

File: src/library/returnBook/ReturnBookControl.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }
+
+        ui.SeTCoMpLeTeD();
+    }
+
+
+    public void dischargeLoan(boolean isDamaged) {
+        if (!state.equals(ControlState.INSPECTING))
+            throw new RuntimeException("ReturnBookControl: cannot call dischargeLoan except in INSPECTING state");
+
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookControl.java]: #PR41 - D5:
Please add brackets to if statement as per CSG 7 (Anywhere curly brackets MAY be used, they MUST be used)

**Review by haynescrossley — CHANGES_REQUESTED** [review]
@elliotmitchum Please see my comments re some minor changes. Please resolve these before we approve the merge.

File: src/library/returnBook/ReturnBookControl.java (hunk 2 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookControl.java]: https://github.com/elliotmitchum/ITC205/pull/41/commits/1774fdc9d206d8411b12829d60f152f49b04b6dc

File: src/library/returnBook/ReturnBookControl.java (hunk 3 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookControl.java]: https://github.com/elliotmitchum/ITC205/pull/41/commits/1774fdc9d206d8411b12829d60f152f49b04b6dc

File: src/library/returnBook/ReturnBookControl.java (hunk 4 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookControl.java]: https://github.com/elliotmitchum/ITC205/pull/41/commits/94485e68568dcdfbafb265738c2ed2b3f645ce28

File: src/library/returnBook/ReturnBookControl.java (hunk 5 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookControl.java]: https://github.com/elliotmitchum/ITC205/pull/41/commits/926ad4cfc751980dbbf826dcd5e1f0580fc61e68

**Comment by elliotmitchum** [comment]
@haynescrossley This has been updated and ready for your review

File: src/library/returnBook/ReturnBookControl.java (hunk 1 above)
Comment (haynescrossley) [reply src/library/returnBook/ReturnBookControl.java]: [Resolved with this commit](https://github.com/elliotmitchum/ITC205/pull/41/commits/1774fdc9d206d8411b12829d60f152f49b04b6dc)

**Review by haynescrossley — APPROVED** [review]
All changes resolved.
Approved to merge @amrmosit

**Comment by amrmosit** [comment]
Origin master is still operational (checked)
@elliotmitchum @haynescrossley

**Pull request merged**